src/data/equiv/basic.lean

```diff
@@ -353,6 +353,29 @@ protected lemma image_compl {α β} (f : equiv α β) (s : set α) :
   f '' sᶜ = (f '' s)ᶜ :=
 set.image_compl_eq f.bijective
 
+@[simp] lemma symm_preimage_preimage {α β} (e : α ≃ β) (s : set β) :
+  e.symm ⁻¹' (e ⁻¹' s) = s :=
```

Comment:
Do we have a lemma that says `f ⁻¹' (g ⁻¹' s) = (f \comp g) ⁻¹' s` or whatever variant of that is true?

Reply:
Yes it's called `preimage_preimage` (as you would expect).  That is, a few characters longer than the current proof.

Reply:
Right, I ask only because `simp` would solve this automatically if `preimage_preimage` were marked `simp`

Reply:
In general I'm not too happy to mark conditional simp lemmas as simp.

Reply:
In what way is `preimage_preimage` conditional?

Reply:
Ah sorry, I was looking at the wrong lemma.  No, `preimage_preimage` is indeed not conditional.  But then I'm not so sure it's a good idea to mark it simp (because it prevents lemmas like `bar (foo ⁻¹' X) ↔ baz X` from triggering).

Reply:
Don't you mean lemmas like `bar ⁻¹'(foo ⁻¹' X) ↔ baz X`? Do we have any of those?

Reply:
Either way, I guess its out of scope for this PR.